Api contract/notify (#165)
Real-Dev-Squad/website-api-contracts: notify/README.md

Reviewer: can we incorporate api versioning as this would be helpful to identify and integrate with other services.

Reviewer: Request you take suggestions from other fellow devs as well.I am ok with `/fcm-token` as token is resource noun but notify sounds more like a verb than noun to me 👀

Author: will notiofication be fine ?

Reviewer: Please format both the file there seems to be some red highlighted lines in both the files indicating the formatting is not proper

# Collection - notify

## API Endpoints

|            Route             |                  Description                  |
| :--------------------------: | :-------------------------------------------: |
| [POST /notify](#post-notify) | send notification to your specified fcm token |
|                              |

## POST /notify

    Send notification to your specified device.

- **Params**  
  None
- **Query**
  None
- **Body**
  - Attributes
    - **title** (required, string): notification title

>     - **body** (required, string): notification message

Reviewer: can we specify allowed character types & maximum lengths considering FCM allows only 4 kb of data

Author: yes we will be giving it in frontend side

Reviewer: We need to have it from backend as well

Author: sure

    - **userId** (string): user who is assigning

>     - **groupRoleId** (string): group to whom they are assigning

Reviewer: can a user send notification to a group only or it can be an individual as as well

Author: individual as well as group too

- **Headers**
  - Content-Type: application/json
- **Cookie**
  None
- **Success Response:**
  - **Code:** 200
    - **Content:**
      ```json
       {  
      "status": 200,
      "message": "User notified successfully"
       }
      ```
- **Error Response:**

  - **Code:** 400

    - **Content:**
      ```json
        {
         "statusCode": 400,
         "error": "Bad Request",
         "message": "\"value\" must contain at least one of [userId, groupRoleId]"
        }
      ```

  - **Code:** 500
    - **Content:**
      ```json
      {
        "message": "The server has encountered an unexpected error. Please contact the administrator for more information."

>       }

Reviewer: key `status code` and `error` are missing here unlike 400 example above

# Collection - no collection for now

## API Endpoints

|              Route              |                  Description                  |
| :-----------------------------: | :-------------------------------------------: |

> | [POST /v1/notify](#post-notify) | send notification to your specified fcm token |

Reviewer 2: isn't notify a **verb**?

Author: giveen `notification`

|                                 |

## POST /v1/notify

    Send notification to your specified device.

- **Params**  
  None
- **Query**
  None
- **Body**

  - Attributes
    - **title** (required, string): notification title
    - **body** (required, string): notification message
    - **userId** (string): user who is assigning

>     - **groupRoleId** (string): group to whom they are assigning

Reviewer 2: ```suggestion
    - **groupRoleId** (string): group to whom the notification will be broadcasted
```

can we do this?

Author: did that

>     - **groupRoleId** (string): group to whom they are assigning

Reviewer 2: can add if this is required or optional?

- **Headers**
  - Content-Type: application/json
- **Cookie**
  None
- **Success Response:**
  - **Code:** 200
    - **Content:**
      ```json
      {
        "status": 200,
        "message": "User notified successfully"
      }
      ```
- **Error Response:**

  - **Code:** 400

    - **Content:**

      ```json
      {
        "statusCode": 400,
        "error": "Bad Request",
        "message": "\"value\" must contain at least one of [userId, groupRoleId]"
      }
      ```

      - **Code:** 401

    - **Content:**
      ```json
      {
        "statusCode": 406,
        "error": "Bad Request",

>         "message": "the message length exceeds"

Reviewer 2: ```suggestion
        "message": "Message length exceeds"
```

      }
      ```

  - **Code:** 500
    - **Content:**
      ```json
      {
        "statusCode": 500,
        "message": "The server has encountered an unexpected error. Please contact the administrator for more information."
      }
      ```

- **Example for device fcm-token document creation request:**
  POST /v1/notify<br/>
  Content-Type: application/json<br/>
  Request-Body:<br/>

>   ```json

Reviewer: formatting
